public/js/p3/widget/templates/UserProfileForm.html

```diff
@@ -21,7 +21,7 @@ <h2 style="margin:0px;padding:4px;font-size:1.2em;text-align:center;background:#
           <tr><td style="padding-top:0">
             <input type="text" name="email" style="width:100%" data-dojo-attach-point="emailField" data-dojo-type="dijit/form/ValidationTextBox" data-dojo-props="placeholder:'Email Address',promptMessage:'Enter your email address',invalidMessage:'Invalid Email Address',validator:dojox.validate.isEmailAddress,trim:true, intermediateChanges: true" required="true" value="" data-dojo-attach-event="onChange: fieldChanged">
           </td><td style="padding-top:0">
-            <input class="useridField" type="text" style="display:block; width:60%" name="username" data-dojo-attach-point="UNF" data-dojo-type="dijit/form/ValidationTextBox" value="" required="true" data-dojo-props="placeholder:'Username',promptMessage:'Enter your username', intermediateChanges: true" data-dojo-attach-event="onChange: fieldChanged">
+            <input class="useridField" type="text" style="display:block; width:60%" name="username" data-dojo-attach-point="UNF" data-dojo-type="dijit/form/ValidationTextBox" value="" required="true" data-dojo-props="regExp:'[A-Za-z0-9_\.]+',invalidMessage:'Username can only contain characters, dijits, underscores, and periods',placeholder:'Username',promptMessage:'Enter your username', intermediateChanges: true" data-dojo-attach-event="onChange: fieldChanged">
```

Comment:
i think i already addressed this. Are you on top of develop branch?

Reply:
https://github.com/PATRIC3/p3_web/commit/79d96636d9278d3f80266b1df081b9fcc0859040